src/common-util/src/main/java/ee/ria/xroad/common/certificateprofile/impl/IsAuthCertificateProfileInfo.java

```diff
@@ -39,21 +39,28 @@
      */
     public IsAuthCertificateProfileInfo(Parameters params) {
         super(new DnFieldDescription[] {
-                // Country Code
+                // Instance identifier
                 new EnumLocalizedFieldDescriptionImpl(
-                        "C", DnFieldLabelLocalizationKey.COUNTRY_CODE, "IS").setReadOnly(true),
+                        "C",
+                        DnFieldLabelLocalizationKey.INSTANCE_IDENTIFIER,
+                        params.getServerId().getXRoadInstance()
+                ).setReadOnly(true),
 
-                // Organization name
+                // Server code
                 new EnumLocalizedFieldDescriptionImpl(
-                        "O", DnFieldLabelLocalizationKey.ORGANIZATION_NAME, "").setReadOnly(false),
+                        "CN",
+                        DnFieldLabelLocalizationKey.SERVER_CODE,
+                        params.getServerId().getServerCode()
+                ).setReadOnly(true),
 
                 // Serialnumber
-                new EnumLocalizedFieldDescriptionImpl("serialNumber", DnFieldLabelLocalizationKey.SERIAL_NUMBER,
-                        params.getServerId().getXRoadInstance() + "/" + params.getServerId().getServerCode()
-                                + "/" + params.getServerId().getMemberClass()).setReadOnly(true),
-
-                // Server code
                 new EnumLocalizedFieldDescriptionImpl(
-                        "CN", DnFieldLabelLocalizationKey.SERVER_DNS_NAME, "").setReadOnly(false) });
+                        "serialNumber",
+                        DnFieldLabelLocalizationKey.SERIAL_NUMBER,
+                        params.getServerId().getXRoadInstance()
+                        + "/" + params.getServerId().getServerCode()
+                        + "/" + params.getServerId().getMemberClass()
```

Comment:
Maybe serialNumber value could be something like ` params.getServerId().getOwner().toShortString()`  if the intention is to make the DN unique (serverCode is already mapped to CN).
Note that serialNumber should be an ASN.1 PrintableString (https://en.wikipedia.org/wiki/PrintableString), and the identifiers (serverId components) can include "non-PrintableString" chars.